### PR TITLE
add loop functionality to CloneEffect / refactor some cards!

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-gui/res/cardsfolder/i/infinite_reflection.txt [hunk 1]
```diff
@@ -2,13 +2,9 @@ Name:Infinite Reflection
...
-SVar:RememberInitialAttach:DB$ Pump | RememberObjects$ Valid Card.AttachedBy
-T:Mode$ ChangesZone | Destination$ Battlefield | ValidCard$ Card.Self+AttachedTo Card | Execute$ FirstReflections | TriggerZones$ Battlefield | TriggerDescription$ When CARDNAME enters the battlefield attached to a creature, each other nontoken creature you control becomes a copy of that creature.
-SVar:FirstReflections:DB$ RepeatEach | RepeatCards$ Creature.nonToken+YouCtrl+IsNotRemembered | UseImprinted$ True | Zone$ Battlefield | RepeatSubAbility$ Reflect | SubAbility$ DBCleanup
-SVar:Reflect:DB$ Clone | Defined$ Remembered | CloneTarget$ Imprinted
-SVar:DBCleanup:DB$ Cleanup | ClearRemembered$ True
+A:SP$ Attach | ValidTgts$ Creature | AILogic$ HighestEvaluation
+T:Mode$ ChangesZone | Destination$ Battlefield | ValidCard$ Card.Self+AttachedTo Card | Execute$ TrigClone | TriggerZones$ Battlefield | TriggerDescription$ When CARDNAME enters the battlefield attached to a creature, each other nontoken creature you control becomes a copy of that creature.
+SVar:TrigClone:DB$ Clone | Defined$ Valid Creature.AttachedBy | CloneTarget$ Valid Creature.nonToken+YouCtrl
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/i/infinite_reflection.txt:7]: this breaks:
> If Infinite Reflection enters the battlefield attached to a creature and then becomes attached to a different creature before its enters-the-battlefield trigger resolves, each other nontoken creature you control will become a copy of the first creature.

File: forge-gui/res/cardsfolder/s/sakashimas_will.txt [hunk 2]
```diff
@@ -1,12 +1,10 @@
...
+SVar:DBControl:DB$ ChooseCard | ValidTgts$ Opponent | Mandatory$ True | Choices$ Creature.TargetedPlayerCtrl | ChoiceTitle$ Choose a creature you control | SubAbility$ DBGainControl | AILogic$ WorstCard | SpellDescription$ Target opponent chooses a creature they control. You gain control of it.
 SVar:DBGainControl:DB$ GainControl | Defined$ ChosenCard | NewController$ You
-SVar:DBImprint:DB$ ChooseCard | Defined$ You | Mandatory$ True | Choices$ Creature.YouCtrl | ChoiceTitle$ Choose a creature you control | SubAbility$ MassClone | SpellDescription$ Choose a creature you control. Each other creature you control becomes a copy of that creature until end of turn.
-SVar:MassClone:DB$ RepeatEach | RepeatCards$ Card.Creature+nonChosenCard+YouCtrl | RepeatSubAbility$ DBCopy
-SVar:DBCopy:DB$ Clone | Defined$ ChosenCard | CloneTarget$ Remembered | Duration$ UntilEndOfTurn
+SVar:DBClone:DB$ Clone | Choices$ Creature.YouCtrl | ChoiceTitle$ Choose a creature you control | ExcludeChosen$ True | CloneTarget$ Valid Creature.YouCtrl
```
Comment (tool4ever) [inline forge-gui/res/cardsfolder/s/sakashimas_will.txt:7]: dropped Duration

File: forge-gui/res/cardsfolder/i/infinite_reflection.txt (hunk 1 above)
Comment (Northmoc) [reply forge-gui/res/cardsfolder/i/infinite_reflection.txt:7]: Ah that explains that static a bit more – thanks

Comment (Northmoc) [reply forge-gui/res/cardsfolder/i/infinite_reflection.txt]: I think this is sorted

**Review by tool4ever — APPROVED** [review]

**Pull request merged**